swim/test_utils.go

```diff
@@ -88,21 +88,39 @@ func (n *testNode) Destroy() {
 	}
 }
 
-func newChannelNode(t *testing.T, address string) *testNode {
+// newChannelNode creates a testNode with a listening channel and associated
+// SWIM node. The channel listens on a random port assigned by the OS.
+func newChannelNode(t *testing.T) *testNode {
 	ch, err := tchannel.NewChannel("test", nil)
 	require.NoError(t, err, "channel must create successfully")
 
-	node := NewNode("test", address, ch.GetSubChannel("test"), nil)
+	// Set the channel listening so it binds to the socket and we get a port
+	// allocated by the OS
+	err = ch.ListenAndServe("127.0.0.1:0")
+	require.NoError(t, err, "channel must listen")
 
-	require.NoError(t, ch.ListenAndServe(node.Address()), "channel must listen")
+	hostport := ch.PeerInfo().HostPort
+	node := NewNode("test", hostport, ch.GetSubChannel("test"), nil)
 
 	return &testNode{node, ch}
 }
 
-func genChannelNodes(t *testing.T, hostports []string) (nodes []*testNode) {
-	for _, hostport := range hostports {
-		node := newChannelNode(t, hostport)
+// newChannelNodeWithHostPort creates a testNode with the address specified by
+// the hostport parameter.
+func newChannelNodeWithHostPort(t *testing.T, hostport string) *testNode {
+	ch, err := tchannel.NewChannel("test", nil)
+	require.NoError(t, err, "channel must create successfully")
+
+	node := NewNode("test", hostport, ch.GetSubChannel("test"), nil)
+
+	return &testNode{node, ch}
+}
+
+func genChannelNodes(t *testing.T, n int) (nodes []*testNode, hostports []string) {
```

Comment:
Is the second return value needed? I don't see it used anywhere. Is the hostport information available from the node itself?